src/compas_timber/elements/beam.py

```diff
@@ -206,6 +216,93 @@ def __str__(self):
             self.frame,
         )
 
+    # ==========================================================================
+    # Implementations of abstract methods
+    # ==========================================================================
+
+    def compute_geometry(self, include_features=True):
+        # type: (bool) -> compas.datastructures.Mesh | compas.geometry.Brep
+        """Compute the geometry of the element.
+
+        Parameters
+        ----------
+        include_features : bool, optional
+            If ``True``, include the features in the computed geometry.
+            If ``False``, return only the base geometry.
+
+        Returns
+        -------
+        :class:`compas.datastructures.Mesh` | :class:`compas.geometry.Brep`
+
+        """
+        blank_geo = Brep.from_box(self.blank)
+        if include_features:
+            for feature in self.features:
+                try:
+                    blank_geo = feature.apply(blank_geo)
+                except FeatureApplicationError as error:
+                    self.debug_infos.append(error)
+        return blank_geo
```

Comment:
that would break the `Element` interface though. previously with the consumer, this information would get created externally. Now with the geometry being calculated internally, it's a bit tricky to get a hold of this information which is necessary for the visual debugging feature of CT.